Sources/Screen/VideoRotator.swift

```diff
@@ -0,0 +1,173 @@
+import CoreImage
+import Foundation
+import ReplayKit
+import VideoToolbox
+
+public enum VideoRotatorError: Error {
+    case noImageBuffer
+    case noOrientationInfo
+    case unsupportedOrientation
+    case cannotAllocatePixelBuffer(CVReturn)
+    case rotationFailure(OSStatus)
+}
+
+public class VideoRotator {
+    private var pixelInfo = PixelInfo.zero {
+        didSet {
+            guard pixelInfo != oldValue else {
+                return
+            }
+            rotationPixelBuffer = nil
+            pixelBufferStatus = CVPixelBufferCreate(
+                kCFAllocatorDefault,
+                pixelInfo.width,
+                pixelInfo.height,
+                pixelInfo.format,
+                nil,
+                &rotationPixelBuffer
+            )
+        }
+    }
+    private var rotationPixelBuffer: CVPixelBuffer?
+    private var pixelBufferStatus: CVReturn = kCVReturnSuccess
+    private let session: VTPixelRotationSession
+
+    public init?() {
+        guard #available(iOS 16.0, tvOS 16.0, macOS 13.0, *) else {
+            return nil
+        }
```

Comment:
[MUST]
- Please mark the class itself with @available(iOS 16.0, tvOS 16.0, macOS 13.0, *) instead of adding it to each method.